### ISSUE TITLE
TSFunction highlights are not being set on first loading nvim

### ISSUE THREAD

**Issue by akinsho** [NTBBloodbath/doom-one.nvim#1]
Hi 👋🏾 ,

Thanks for working on this. I'd been hoping for a lua based alternative to the vimscript colorscheme since that was contributing quite a lot to my startup time.

One thing I've noticed since moving over though is that function highlights aren't applied until something else is triggered. Initially highlights for `TSFunction` aren't being coloured correctly despite the `Function` highlight group showing up correctly.

![image](https://user-images.githubusercontent.com/22454918/121718306-13df1380-cada-11eb-8d6a-df518c6c037a.png)

when I print the value of the highlight groups I see
![image](https://user-images.githubusercontent.com/22454918/121718442-3b35e080-cada-11eb-8e4a-cdac8b29f893.png)

and `Function` shows

![image](https://user-images.githubusercontent.com/22454918/121718555-56085500-cada-11eb-9ab3-cd94ab6bd498.png)

so they seem to be set correctly but aren't highlighted in the buffer.

The minimal `init.lua` I can reproduce this with is
```lua
vim.cmd("packadd! doom-one.nvim")
vim.cmd("packadd nvim-treesitter")

vim.cmd("syntax enable")

vim.o.termguicolors = true

vim.cmd [[colorscheme doom-one]]

require("nvim-treesitter.configs").setup {
  highlight = {
    enable = true
  }
}
```

Just as a side note I've noticed that when I open a `neogit` buffer or use `TSHighlightCapturesUnderCursor` from `treesitter-playground` to investigate they suddenly appear. I'm sure it isn't really about those specific plugins but maybe something async is happening or something waiting on an autocommand maybe?

**Comment by NTBBloodbath** [comment]
Hey, thanks for reporting and adding that information, will be really useful!

I'll see what I can do right now :slightly_smiling_face:

**Comment by NTBBloodbath** [comment]
After a while I realized that there were groups of TS cleared so I fixed them. It should be solved by now, let me know if you find any other problems!

**Comment by akinsho** [comment]
@NTBBloodbath thanks for the fix that now works well 👍🏾 , the colour for highlighting methods has changed to orange now though, not sure if that was intentional?

**Comment by NTBBloodbath** [comment]
> the colour for highlighting methods has changed to orange now though, not sure if that was intentional?

Yeah, I made the Property group orange. Lemme know if other color will look better :)

**Comment by akinsho** [comment]
@NTBBloodbath tbh I get that with colorschemes it's all a matter of personal preference anyway and tbh  can just overwrite parts of it in my config. In the screenshot below from the readme they are a light purple.

![image](https://user-images.githubusercontent.com/22454918/121811315-ede17c80-cc5b-11eb-8444-44c0044a113b.png)

Tbh I think first and foremost the theme should be faithful to how things look in doom. I couldn't find an example of a method in a doom screenshot (and I don't actually use doom just like the theme), so if the current appearance is what doom uses then I'm fine with it being faithful to that and can overwrite it in my config. Personally I think the light purple or even the yellow or some softer colour would be nicer than the orange.

It's quite jarring compared to the rest of the palette. It makes sense for some things like the booleans which I quite like being able to immediately spot a boolean but if you use a language with a lot of properties (like OOP languages) then a lot of the screen is bright orange 🤷🏾 

Again it's easily solvable on my end just my 2 cents.
